tensorboard/backend/application.py

```diff
@@ -91,6 +92,13 @@ def TensorBoardWSGIApp(
       auth_providers: Optional mapping whose values are `AuthProvider` values
         and whose keys are used by (e.g.) data providers to specify
         `AuthProvider`s via the `AuthContext.get` interface. Defaults to `{}`.
+      middlewares: Optional list of WSGI middlewares (i.e., callables that take
```

Comment:
Maybe it would be prudent to explicitly mark this parameter as "experimental" for now?  The `TensorBoardWSGIApp` API is still basically a private API, but I think the rest of the signature is a bit more established so it might be worth differentiating this one, given that based on internal conversation it doesn't seem like we've fully settled how we want to manage the middleware tower in the long term.

(E.g. I could imagine that at some point perhaps we'd rather have extra routes implemented via an actual `extra_routes` parameter taking a mapping of path to WSGI app, so that would no longer require passing `middlewares` - or we could decide to peel off the outer middlewares from the return value of this function and instead require callers to apply those "by hand" at which point this becomes obsolete.)

Reply:
Sure, that’s fine with me. Agreed that I’m not sure exactly what the
right form here is. Done.